src/desktop.c

```diff
@@ -2298,13 +2330,20 @@ static void update_background(FmDesktop* desktop, int is_it)
             }
         }
 #if GTK_CHECK_VERSION(3, 0, 0)
-        xdisplay = GDK_WINDOW_XDISPLAY(root);
-        /* this code is taken from libgnome-desktop */
-        xpixmap = XCreatePixmap(xdisplay, RootWindow(xdisplay, screen_num),
-                                dest_w, dest_h, DefaultDepth(xdisplay, screen_num));
-        cache->bg = cairo_xlib_surface_create(xdisplay, xpixmap,
-                                              GDK_VISUAL_XVISUAL(gdk_screen_get_system_visual(screen)),
-                                              dest_w, dest_h);
+        if(IS_X11())
+        {
+#ifdef HAVE_X11
```

Comment:
Makes no sense at all. If `HAVE_X11` undefined then `IS_X11` is `FALSE`, i.e. at this point `HAVE_X11` is 100% defined.

Reply:
The #ifdef's inside the if(IS_X11()) make sense, because else you get compiler errors due to it using X11-specific APIs that you won't have.

Reply:
Well, in such case enclose all `if`...`else` into `#ifdef` please.